client/src/components/panels/TickPanel.tsx

```
@@ -71,6 +73,21 @@ export function TickPanel({
     (window as unknown as { setReady?: () => void }).setReady?.();
   };
 
+  // Keyboard shortcut handler for Shift+E (End Turn)
+  useEffect(() => {
+    if (gameMode !== 'single_player' || isTestMode) return;
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)